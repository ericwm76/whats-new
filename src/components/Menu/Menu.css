.menu {
  align-items: center;
  display: flex;
  flex-direction: column;
  grid-area: menu;
  /* margin-top: 50px; */
}

.menu-btn {
  background-color: #FF964C;
  height: 40px;
  width: 90%;
  margin: 5px;
  border-radius: 5px;
}