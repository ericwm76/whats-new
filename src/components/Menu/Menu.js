import React from 'react';
import './Menu.css'

const Menu = ({ changeSelectedNews }) => {
  return (
    <section className="menu">
      <button className="menu-btn" name="local" onClick={ e => changeSelectedNews(e) }>Local</button>
      <button className="menu-btn" name="science" onClick={ e => changeSelectedNews(e) }>Science</button>
      <button className="menu-btn" name="technology" onClick={ e => changeSelectedNews(e) }>Technology</button>
      <button className="menu-btn" name="health" onClick={ e => changeSelectedNews(e) }>Health</button>
      <button className="menu-btn" name="entertainment" onClick={ e => changeSelectedNews(e) }>Entertainment</button>
    </section>
  )
} 


export default Menu;