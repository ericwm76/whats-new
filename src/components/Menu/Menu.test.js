import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () =>{
  let wrapper;
  const mockChangeSelectedNews = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Menu changeSelectedNews={mockChangeSelectedNews}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should run changeSelectedNews when Local button is clicked', () => {
    wrapper.find('.menu-btn').at(0).simulate('click');
    expect(mockChangeSelectedNews).toHaveBeenCalled();
  })

  it('should run changeSelectedNews when Science button is clicked', () => {
    wrapper.find('.menu-btn').at(1).simulate('click');
    expect(mockChangeSelectedNews).toHaveBeenCalled();
  })

  it('should run changeSelectedNews when Technology button is clicked', () => {
    wrapper.find('.menu-btn').at(2).simulate('click');
    expect(mockChangeSelectedNews).toHaveBeenCalled();
  })

  it('should run changeSelectedNews when Health button is clicked', () => {
    wrapper.find('.menu-btn').at(3).simulate('click');
    expect(mockChangeSelectedNews).toHaveBeenCalled();
  })

  it('should run changeSelectedNews when Entertainment button is clicked', () => {
    wrapper.find('.menu-btn').at(4).simulate('click');
    expect(mockChangeSelectedNews).toHaveBeenCalled();
  })
})