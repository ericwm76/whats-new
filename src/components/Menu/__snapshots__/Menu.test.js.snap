// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<section
  className="menu"
>
  <button
    className="menu-btn"
    name="local"
    onClick={[Function]}
  >
    Local
  </button>
  <button
    className="menu-btn"
    name="science"
    onClick={[Function]}
  >
    Science
  </button>
  <button
    className="menu-btn"
    name="technology"
    onClick={[Function]}
  >
    Technology
  </button>
  <button
    className="menu-btn"
    name="health"
    onClick={[Function]}
  >
    Health
  </button>
  <button
    className="menu-btn"
    name="entertainment"
    onClick={[Function]}
  >
    Entertainment
  </button>
</section>
`;
